### PR TITLE
sui: uptick to version 0.29

### LINKED ISSUE #2499: sui: address and ID changing from 20 bytes to 32 bytes
## Description and context

According to a breaking change for Sui 0.28, we will need to review the contracts to make sure we handle `address` and `ID` as 32-byte types because they will no longer be 20 bytes. For example, `Bytes32` custom type should just be able to convert to and from `address` without requiring `Bytes20` to assist.

## Definition of done

- [ ] Review and fix anywhere Wormhole contract is affected
- [ ] Review and fix anywhere Token Bridge contract is affected

## References

https://docs.sui.io/doc-updates/sui-breaking-changes

### LINKED ISSUE #2501: sui: fix signature verification logic to pass in raw message
## Description and context

According to a breaking change for Sui 0.28, we will need to fix signature verification to pass in the raw message (single keccak256 hash of message body) instead of double keccak256. We may want to store the hash in `VAA` differently (currently storing the double hash).

Keep in mind that signature verification using `ecdsa_k1` will now require an additional argument (`u8`) specifying the hashing algorithm. In our case we will use `0` (for keccak256).

## Definition of done

- [ ] Fix and test signature verification in `guardian_set` module.

## References

https://github.com/MystenLabs/sui/pull/7773
https://docs.sui.io/doc-updates/sui-breaking-changes

### PR DESCRIPTION
**Pull request by a5-pickle**
Closes #2499.
Closes #2501.

### CODE REVIEW

**Comment by a5-pickle** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2358** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2358" target="_blank">Graphite</a> 
    * **PR #2551** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2551" target="_blank">Graphite</a>  👈
    * **PR #2516** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2516" target="_blank">Graphite</a> 
    * **PR #2494** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2494" target="_blank">Graphite</a> 
    * **PR #2534** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2534" target="_blank">Graphite</a> 
      * **PR #2547** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2547" target="_blank">Graphite</a> 
        * **PR #2549** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2549" target="_blank">Graphite</a> 
    * **PR #2543** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2543" target="_blank">Graphite</a> 
      * **PR #2544** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2544" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2551?utm_source=stack-comment).

File: sui/token_bridge/Move.toml [hunk 1]
```diff
@@ -4,8 +4,8 @@ version = "0.0.1"
 
 [dependencies.Sui]
 git = "https://github.com/MystenLabs/sui.git"
-subdir = "crates/sui-framework"
-rev = "157ac72030d014f17d76cefe81f3915b4afab2c9"
+subdir = "crates/sui-framework/packages/sui-framework"
```
Comment (optke3) [inline sui/token_bridge/Move.toml:7]: this is what messed me up earlier

**Review by optke3 — APPROVED** [review]
Nice job!

**Review by heyitaki — APPROVED** [review]

**Review by optke3 — APPROVED** [review]
approved

**Pull request merged**